echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/eventhandlers/ManualEventHandler.java

```diff
@@ -93,4 +102,23 @@ private Pipeline buildTrigger(Pipeline pipeline, Trigger manualTrigger) {
     }
     return notifications;
   }
+
+  // Manual triggers try to replicate actual events (and in some cases build events) but rather than pass the event to
+  // echo, they add the information to the trigger. It may make sense to refactor manual triggering to pass a trigger and
+  // an event as with other triggers, but for now we'll see whether we can extract a build event from the trigger.
+  private Optional<BuildEvent> extractBuildInformation(Trigger manualTrigger) {
+    String master = manualTrigger.getMaster();
+    String job = manualTrigger.getJob();
+    if (StringUtils.isNoneEmpty(master, job)) {
```

Comment:
What is `isNoneEmpty`?

Reply:
It's equivalent to `StringUtils.isNotEmpty(master) && StringUtils.isNotEmpty(job)` but shorter...though maybe it's clearer just to just `isNotEmpty`?

Reply:
I think this is reasonable, just wondering.